src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -130,13 +175,15 @@ The .NET Foundation licenses this file to you under the MIT license.
       <LinkerArg Include="-gz=zlib" Condition="'$(CompressSymbols)' != 'false'" />
       <LinkerArg Include="-fuse-ld=$(LinkerFlavor)" Condition="'$(LinkerFlavor)' != ''" />
       <LinkerArg Include="@(NativeLibrary)" />
-      <LinkerArg Include="--sysroot=$(SysRoot)" Condition="'$(SysRoot)' != ''" />
-      <LinkerArg Include="--target=$(TargetTriple)" Condition="'$(_IsApplePlatform)' != 'true' and '$(TargetTriple)' != ''" />
+      <LinkerArg Include="--sysroot=$(SysRoot)" Condition="'$(SysRoot)' != '' and '$(_iOSLibraryMode)' != 'true'" />
+      <LinkerArg Include="-isysroot $(SysRoot)" Condition="'$(SysRoot)' != '' and '$(_iOSLibraryMode)' == 'true'" />
```

Comment:
@jkotas regarding our discussion in https://github.com/dotnet/runtime/pull/94999#discussion_r1420504285 is there anything you would like me to address in this PR?

Reply:
I do not think it makes sense for `SysRoot` to mean SDK root (isysroot) when building iOS-like libraries and a classic SysRoot for all other Apple targets.

I think overriding SDK root is an advanced scenario for all Apple targets. If we want `SysRoot` on Apple platforms to mean SDK root, this condition should be `_IsApplePlatform`.

(Alternatively, if we want to keep `SysRoot` as a classic SysRoot on Apple platforms, we should invent a property with a different name for the SDK root. Again, it should be applicable to all Apple platforms.)

Reply:
Thank you, address in the latest commit.